libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2688,11 +2690,14 @@ vector<VariableDeclaration const*> SMTEncoder::modifiersVariables(FunctionDefini
 
 		visited.insert(modifier);
 		solAssert(_contract, "No contract context provided for modifier analysis!");
-		auto const& actualModifier = modifier->resolveVirtual(*_contract);
-		if (actualModifier.isImplemented())
+		VirtualLookup lookup = *invok->name().annotation().requiredLookup;
+		solAssert(lookup == VirtualLookup::Virtual || lookup == VirtualLookup::Static, "");
+		if (lookup == VirtualLookup::Virtual)
+			modifier = &modifier->resolveVirtual(*_contract);
```

Comment:
I think it's worth extracting the 4 lines above into a function, and reuse that also in the other modified code.